connector/src/main/scala/com/microsoft/kusto/spark/utils/KustoClient.scala

```diff
@@ -167,10 +167,39 @@ class KustoClient(val clusterAlias: String, val engineKcsb: ConnectionStringBuil
   private[kusto] def cleanupIngestionByproducts(database: String, kustoAdminClient: Client, tmpTableName: String): Unit = {
     try {
       kustoAdminClient.execute(database, generateTableDropCommand(tmpTableName))
+      KDSU.logInfo(myName, s"Temporary table '$tmpTableName' deleted successfully")
     }
     catch {
       case exception: Exception =>
-        KDSU.reportExceptionAndThrow(myName, exception, s"deleting temporary table $tmpTableName", database = database, shouldNotThrow = true)
+        KDSU.reportExceptionAndThrow(myName, exception, s"deleting temporary table $tmpTableName", database = database, shouldNotThrow = false)
+    }
+  }
+
+  def cleanupTempTables(coordinates: KustoCoordinates): Unit = {
+    Future {
+      val tempTablesOld: Seq[String] =
+        engineClient.execute(generateFindOldTempTablesCommand(coordinates.database))
+          .getPrimaryResults.getData.asInstanceOf[util.ArrayList[util.ArrayList[String]]].asScala
+          .headOption.map(_.asScala)
+          .getOrElse(Seq())
+
+      // Try delete temporary tablesToCleanup created and not used
+      val tempTablesCurr = engineClient.execute(coordinates.database, generateFindCurrentTempTablesCommand(TempIngestionTablePrefix)).getPrimaryResults
+      if (tempTablesCurr.count() > 0) {
+        tempTablesCurr.next()
+        val tablesToCleanup: Seq[String] = tempTablesOld.intersect(tempTablesCurr.getCurrentRow.asScala)
+        if (tablesToCleanup.nonEmpty) {
+          engineClient.execute(coordinates.database, generateDropTablesCommand(tablesToCleanup.mkString(",")))
+        }
+      }
+    } onFailure {
+      case ex: Exception =>
+        KDSU.reportExceptionAndThrow(
+          myName,
+          ex,
+          "trying to drop temporary tables", coordinates.clusterUrl, coordinates.database, coordinates.table.getOrElse("Unspecified table name"),
```

Comment:
Don't you want to emit an explicit error for each table that we failed to delete?
Like error per table with full details of the table that failed to be deleted.